avocado/core/runners/sysinfo.py

```diff
@@ -0,0 +1,298 @@
+import logging
+import multiprocessing
+import os
+import time
+import traceback
+from abc import ABC, abstractmethod
+
+from avocado.core import nrunner
+from avocado.core.runners.utils import messages
+from avocado.core.settings import settings
+from avocado.utils import astring, process, software_manager
+
+__all__ = ["PreSysInfo", "PostSysInfo", "SysinfoRunner", "RunnerApp", "main"]
+
+log = logging.getLogger("avocado.sysinfo")
+
+
+class Collectible(ABC):
+
+    """
+    Abstract class for representing collectibles by sysinfo.
+    """
+
+    def __init__(self, log_path):
+        self.log_path = astring.string_to_safe_path(log_path)
+
+    def __eq__(self, other):
+        if hash(self) == hash(other):
+            return True
+        elif isinstance(other, Collectible):
+            return False
+        return NotImplemented
+
+    def __ne__(self, other):
+        result = self.__eq__(other)
+        if result is NotImplemented:
+            return result
+        return not result
+
+    @abstractmethod
+    def run(self, queue, logdir):
+        pass
+
+
+class LogFile(Collectible):
+
+    """
+    Collectible system file.
+
+    :param path: Path to the log file.
+    """
+
+    def __init__(self, path):
+        super().__init__(os.path.basename(path))
+        self.path = path
+
+    def __repr__(self):
+        return "LogFile(%r, %r)" % (self.path, self.log_path)
+
+    def __hash__(self):
+        return hash((self.path, self.log_path, LogFile))
+
+    def run(self, queue, logdir):
+        """
+        Send the content of log file to the runner messages queue.
+
+        :param queue: queue for the runner messages
+        :type queue: multiprocessing.SimpleQueue
+        :param logdir: Path to a log directory.
+        :type logdir: str
+        """
+        if os.path.exists(self.path):
+            try:
+                with open(self.path) as f:
+                    lines = f.readlines()
+                log_path = os.path.join(logdir, self.log_path)
+                for line in lines:
+                    queue.put(messages.FileMessage.get(line, log_path))
```

Comment:
Why is this sending line by line? It puts a lot of overhead on all the "envelopes" for each line, at all levels.  This will be read by users *after* they are collected in 99.99% percent of use cases (maybe even more).